### ISSUE TITLE
3D Logic 2: script hangs

### ISSUE THREAD

**Issue by n0samu** [ruffle-rs/ruffle#6086]
### Describe the bug

SWF URL: http://i.notdoppler.com/files/3dlogic2.swf
After clicking the "New Game" button on the main menu, Ruffle hangs. Eventually I get this message in the console: 
> A script in this movie has taken too long to execute and has been terminated.
> No more actions will be executed in this movie.

### Expected behavior

The main menu should fade out and the game should start.

### Affected platform

Browser's extension

### Operating system

Windows 10

### Browser

Mozilla Firefox 96

**Comment by Herschel** [comment]
After clicking new game, the intro text "In a distant swamp..." is set up using a loop based on `TextField.maxscroll`:
```
while(i <= parent.level)
{
   if(parent.story[i] != undefined)
   {
      if(textData != "")
      {
         textData += "\n";
      }
      textData += parent.story[i];
      mcStory._txt.text += parent.story[i];
      while(true)
      {
         mcStory._txt.text += "\n";
         if(mcStory._txt.maxscroll != 1)
         {
            break;
         }
         textData += "\n";
      }
      mcStory._txt.text = "";
   }
   i++;
}
```


 This loop never exits on Ruffle, so probably our maxscroll implementation is buggy (doesn't consider newlines at the end of text?)

**Comment by n0samu** [comment]
Seems like this was fixed by #8301.
